lib/Doctrine/DBAL/Driver/PDOStatement.php

```diff
@@ -79,10 +82,15 @@ public function bindValue($param, $value, $type = ParameterType::STRING)
      */
     public function bindParam($column, &$variable, $type = ParameterType::STRING, $length = null, $driverOptions = null)
     {
-        $type = $this->convertParamType($type);
+        $type            = $this->convertParamType($type);
+        $extraParameters = array_slice(func_get_args(), 3);
+
+        if (count($extraParameters) !== 0) {
+            $extraParameters[0] = $extraParameters[0] ?? 0;
```

Comment:
This is very convoluted/hard to read. You are replacing the 4th parameter with `0` if it is set and it is `null`? What happens if `$extraParameters[0]` is always defined by us (instead of doing the `count()` check)?

Reply:
This comes usually as always-defined, defaulting to null, due to:
https://github.com/doctrine/dbal/blob/e39bc0859fbf6e9bf78e30c4dd3329515c6deccc/lib/Doctrine/DBAL/Statement.php#L110-L129

Reply:
So is `if (count($extraParameters) !== 0) {` needed at all?

Reply:
Someone still may call this API directly, I guess.

Reply:
Yes, I understand that, but the code you provided works with/without `count() !== 0` check, no?

Reply:
No, only when the parameter was really explicitly passed: https://3v4l.org/KORCg
This method is really dirty.

Reply:
I don't really see how we can improve here. Unlike the userland optional arguments defaulting to some value, the internal arguments are counted, so we have to respect the expected number of arguments being passed and their types.

Reply:
Another approach might be to declare `int $length = 0` since on the driver level there's no `null` anyways. But it's fine if we do it later since it's an API change.

Reply:
@Ocramius unless you still have objections and can propose a better approach, I'm going to merge this.

Reply:
Good to go 👍